app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
I'm not sure this is the right logic. If I come in via LOA 3 and I don't already have an account, I get redirected to the IdV flow before I get a recovery code. This is the sequence of events:
1. Make LOA3 request from SP
2. Click "Create account"
3. Enter email and confirm it
4. Enter phone and confirm it
=> You are sent to IdV and you don't have a recovery code yet.

If you cancel IdV, it thinks you want to cancel your account.

I think we can drop the `&& current_user.recovery_code.present?` since all we care about is whether or not this is an LOA3 context.